packages/react/src/components/TreeView/TreeView.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, match, matches } from '../../internal/keyboard';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function TreeView({
+  active: prespecifiedActive,
+  children,
+  className,
+  hideLabel = false,
+  label,
+  multiselect,
+  onSelect,
+  selected: preselected = [],
+  size = 'default',
+  ...rest
+}) {
+  const { current: treeId } = useRef(rest.id || uniqueId());
+  const treeClasses = classNames(className, `${prefix}--tree`, {
+    [`${prefix}--tree--${size}`]: size !== 'default',
+  });
+  const treeRootRef = useRef(null);
+  const treeWalker = useRef(treeRootRef?.current);
```

Comment:
Would `treeRootRef` be initialized by this point yet? Seems like it'd be the same as `useRef(null)` right?

Reply:
yeah functionally they would be the same I think